prime-router/src/main/kotlin/azure/DatabaseAccess.kt

```diff
@@ -190,10 +190,12 @@ class DatabaseAccess(private val create: DSLContext) {
         receiverName: String,
     ): List<Header> {
         val cond = if (since == null) {
-            TASK.RECEIVER_NAME.like("$receiverName%")
+            TASK.SENT_AT.isNotNull()
+                .and(TASK.RECEIVER_NAME.like("$receiverName%"))
         } else {
```

Comment:
For the next round of work:  The "like" is a security flaw.   Once we switch to the new lineage tables, the organization and service are broken apart as separate fields, and you can do a simple exact match.